lib/iris/_merge.py

```diff
@@ -1068,6 +1069,27 @@ def derive_space(groups, relation_matrix, positions, function_matrix=None):
     return space
 
 
+def _multidim_daskstack(stack):
+    """
+    Recursively build a multidensional stacked dask array.
+
+    The argument is an ndarray of dask arrays.
+    This is needed because dask.array.stack only accepts a 1-dimensional list.
+
+    """
+    if stack.ndim == 0:
+        # A 0-d array cannot be merged.
+        result = stack.item()
```

Comment:
I'm still not totally sure this bit is right.
Surely for recursive calling we always need to return a dask object, so we can stack several of those ??
See call for more research [above](https://github.com/SciTools/iris/pull/2369/files#r101527126).